src/System.IO.MemoryMappedFiles/tests/MemoryMappedFile.CreateFromFile.Tests.cs

```diff
@@ -0,0 +1,810 @@
+            using (TempFile file = new TempFile(GetTestFilePath()))
+            {
+                Assert.Throws<ArgumentException>(() => MemoryMappedFile.CreateFromFile(file.Path, FileMode.Truncate));
+            }
+        }
+
+        /// <summary>
+        /// Tests invalid arguments to the CreateFromFile access parameter.
+        /// </summary>
+        [Fact]
+        public void InvalidArguments_Access()
+        {
+            // Out of range access values with a path
+            Assert.Throws<ArgumentOutOfRangeException>("access", () => MemoryMappedFile.CreateFromFile(GetTestFilePath(), FileMode.Open, CreateUniqueMapName(), 4096, (MemoryMappedFileAccess)(-2)));
+            Assert.Throws<ArgumentOutOfRangeException>("access", () => MemoryMappedFile.CreateFromFile(GetTestFilePath(), FileMode.Open, CreateUniqueMapName(), 4096, (MemoryMappedFileAccess)(42)));
+
+            //  Write-only access is not allowed on maps (only on views)
+            Assert.Throws<ArgumentException>("access", () => MemoryMappedFile.CreateFromFile(GetTestFilePath(), FileMode.Open, CreateUniqueMapName(), 4096, MemoryMappedFileAccess.Write));
+
+            // Test the same things, but with a FileStream instead of a path
+            using (TempFile file = new TempFile(GetTestFilePath()))
+            using (FileStream fs = File.Open(file.Path, FileMode.Open))
+            {
+                // Out of range values with a stream
+                Assert.Throws<ArgumentOutOfRangeException>("access", () => MemoryMappedFile.CreateFromFile(fs, CreateUniqueMapName(), 4096, (MemoryMappedFileAccess)(-2), HandleInheritability.None, true));
+                Assert.Throws<ArgumentOutOfRangeException>("access", () => MemoryMappedFile.CreateFromFile(fs, CreateUniqueMapName(), 4096, (MemoryMappedFileAccess)(42), HandleInheritability.None, true));
+
+                // Write-only access is not allowed
+                Assert.Throws<ArgumentException>("access", () => MemoryMappedFile.CreateFromFile(fs, CreateUniqueMapName(), 4096, MemoryMappedFileAccess.Write, HandleInheritability.None, true));
+            }
+        }
+
+        /// <summary>
+        /// Tests various values of FileAccess used to construct a FileStream and MemoryMappedFileAccess used
+        /// to construct a map over that stream.  The combinations should all be valid.
+        /// </summary>
+        [Theory]
+        [InlineData(FileAccess.ReadWrite, MemoryMappedFileAccess.Read)]
+        [InlineData(FileAccess.ReadWrite, MemoryMappedFileAccess.ReadWrite)]
+        [InlineData(FileAccess.ReadWrite, MemoryMappedFileAccess.CopyOnWrite)]
+        [InlineData(FileAccess.Read, MemoryMappedFileAccess.Read)]
+        [InlineData(FileAccess.Read, MemoryMappedFileAccess.CopyOnWrite)]
+        public void FileAccessAndMapAccessCombinations_Valid(FileAccess fileAccess, MemoryMappedFileAccess mmfAccess)
+        {
+            const int Capacity = 4096;
```

Comment:
I went back and forth on that.  DefaultCapacity means something specific in this library, 0 being passed to mean "use the size of the underlying file or map", so I didn't want to use that, and I couldn't come up with something better.  Plus, my choices in many of these cases were largely random, any value would do as long as it was consistent within the method, hence my defining a constant in each of the methods where I needed consistency within that method.  I think I'll leave it this way for now... we can always choose to unify them later if desired.